mlir/lib/Dialect/GPU/IR/GPUDialect.cpp

```diff
@@ -1729,28 +1729,46 @@ LogicalResult gpu::ReturnOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// SpatialExtentsAttr
+//===----------------------------------------------------------------------===//
+
+LogicalResult
+SpatialExtentsAttr::verify(function_ref<InFlightDiagnostic()> emitError,
+                           std::optional<unsigned>,
+                           std::optional<DenseI32ArrayAttr> reqdWorkgroupSize,
+                           std::optional<DenseI32ArrayAttr> maxWorkgroupSize) {
+  if (reqdWorkgroupSize.has_value() && maxWorkgroupSize.has_value())
+    return emitError()
+           << "cannot have both a reqdWorkgroupSize and a maxWorkgroupSize";
+  return success();
+}
+
 //===----------------------------------------------------------------------===//
 // GPUModuleOp
 //===----------------------------------------------------------------------===//
 
-void GPUModuleOp::build(OpBuilder &builder, OperationState &result,
+void GPUModuleOp::build(OpBuilder &odsBuilder, OperationState &odsState,
                         StringRef name, ArrayAttr targets,
-                        Attribute offloadingHandler) {
-  ensureTerminator(*result.addRegion(), builder, result.location);
+                        Attribute offloadingHandler,
+                        SpatialExtentsAttr spatialExtents) {
+  ensureTerminator(*odsState.addRegion(), odsBuilder, odsState.location);
 
-  Properties &props = result.getOrAddProperties<Properties>();
+  Properties &props = odsState.getOrAddProperties<Properties>();
   if (targets)
     props.targets = targets;
-  props.setSymName(builder.getStringAttr(name));
+  props.setSymName(odsBuilder.getStringAttr(name));
   props.offloadingHandler = offloadingHandler;
+  props.spatialExtents = spatialExtents;
 }
 
-void GPUModuleOp::build(OpBuilder &builder, OperationState &result,
+void GPUModuleOp::build(OpBuilder &odsBuilder, OperationState &odsState,
                         StringRef name, ArrayRef<Attribute> targets,
-                        Attribute offloadingHandler) {
-  build(builder, result, name,
-        targets.empty() ? ArrayAttr() : builder.getArrayAttr(targets),
-        offloadingHandler);
+                        Attribute offloadingHandler,
+                        SpatialExtentsAttr spatialExtents) {
+  build(odsBuilder, odsState, name,
+        targets.empty() ? ArrayAttr() : odsBuilder.getArrayAttr(targets),
+        offloadingHandler, spatialExtents);
 }
```

Comment:
Revert renaming

Reply:
this matches the functions signature, but sure it's not worth changing at present.